### PR TITLE
Don't process escape sequences in group names.

### PR DESCRIPTION
**Pull request by AltSysrq**
Supercedes https://github.com/fullcontact/ez-vcard/pull/3, since this
actually can work.

This needs some extra review on that method as a whole; they way it processes escapes in-line rather than extracting the strings then processing them makes this awkward and rather error-prone.

@blendmaster

### CODE REVIEW

File: src/main/java/ezvcard/io/text/VCardRawReader.java [hunk 1]
```diff
@@ -262,12 +267,14 @@ public VCardRawLine readLine() throws IOException {
 			if (ch == '\\' || (ch == '^' && version != VCardVersion.V2_1 && caretDecodingEnabled)) {
 				//an escape character was read
 				escapeChar = ch;
+                                unescapedBuffer.append(ch);
```
Comment (blendmaster) [inline src/main/java/ezvcard/io/text/VCardRawReader.java]: indented wrong?

Comment (AltSysrq) [inline src/main/java/ezvcard/io/text/VCardRawReader.java]: Argh, my line is indented with spaces instead of hard tabs

**Comment by blendmaster** [comment]
SLISW. Worth extra commentary on why group names are read unescaped.

**Comment by AltSysrq** [comment]
Addressed both issues. Also expanded the test to ensure the group buffer is properly cleared before reading the next group.

**Pull request merged**